sdk/client.py

```diff
@@ -82,6 +85,11 @@ def _request(
         data: JSendPayload = rsp.json()
         # If account returns unauthorized we attempt to login and retry request
         if rsp.status_code == 401:
+            #  TODO: Improve the interactive login flow
+            if (self.username == "" and self.password == "") and self.login_url:
```

Comment:
Instead of empty string should we make username and password optional with default values to None ? This check could be replaced by `if not self.username and not self.password`

If we go this way we could also add an error message to the user if all of login_url, user_name and password are falsy

Reply:
I think empty string is also `False`y

Reply:
That's correct I believe! My main point was around defining default values so that we wouldn't need to do username="" when initializing the SDK with the login_url

Reply:
exactly, and having type consistency there would be a bit easier for me downstreams